src/Symfony/Component/EventDispatcher/Debug/NamedListener.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\EventDispatcher\Debug;
+
+/**
+ * The NamedListener marks listener that are able to provide naming for themselves.
+ *
+ * @author Quentin Devos <quentin@devos.pm>
+ */
+interface NamedListener
```

Comment:
I don't see this type being used anywhere. This might be a hint that we don't need it, and neither might we need the abstract class, if we turn this into a trait. Would that work for you?

Reply:
It is used in the Abstract class (L62) to check if the inner listener implements the interface so we can call the function. If we can keep the feature while turning the abstract to a trait, that's also fine for me 👍